app/src/main/java/com/globant/weatherapp/activities/WelcomeActivity.kt

```diff
@@ -7,19 +7,33 @@ import com.globant.weatherapp.mvp.contracts.WeatherContracts
 import com.globant.weatherapp.mvp.model.WeatherModel
 import com.globant.weatherapp.mvp.presenter.WeatherPresenter
 import com.globant.weatherapp.mvp.view.WeatherView
+import kotlinx.android.synthetic.main.activity_welcome_layout.*
 
-class MainActivity : AppCompatActivity() {
+class WelcomeActivity : AppCompatActivity() {
 
     private lateinit var presenter: WeatherContracts.Presenter
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_city_layout)
+        setContentView(R.layout.activity_welcome_layout)
+
         val jsonToString = applicationContext.assets.open(FILE_NAME).bufferedReader().use {
             it.readText()
         }
+
         presenter = WeatherPresenter(WeatherModel(jsonToString), WeatherView(this))
         presenter.initPresenter()
+
+        setOnClickListenerToSearchButton()
+    }
+
+    private fun setOnClickListenerToSearchButton() {
+        activity_welcome_button_search.setOnClickListener {
+            val cityId =
```

Comment:
this is not fine. here show should call presenter.buttonClicked or something like that

Reply:
Done!